File: core/core-api/src/main/java/io/say/better/domain/solution/application/converter/SolutionSymbolConverter.java
package io.say.better.domain.solution.application.converter;

import java.util.List;

import io.say.better.storage.mysql.domain.entity.Solution;
import io.say.better.storage.mysql.domain.entity.SolutionSymbol;
import io.say.better.storage.mysql.domain.entity.Symbol;

public class SolutionSymbolConverter {

	private SolutionSymbolConverter() {
		throw new IllegalStateException("Utility class");
	}

	public static List<SolutionSymbol> toSolutionSymbols(Solution savedSolution, List<Symbol> symbols) {
		return symbols.stream()
				.map(symbol -> toSolutionSymbol(savedSolution, symbol))
				.toList();
	}

	private static SolutionSymbol toSolutionSymbol(Solution solution, Symbol symbol) {
		return SolutionSymbol.builder()
				.solution(solution)
				.symbol(symbol)
				.build();
	}

}

File: core/core-api/src/main/kotlin/io/say/better/domain/solution/application/converter/SolutionSymbolConverter.kt
package io.say.better.domain.solution.application.converter

import io.say.better.storage.mysql.domains.solution.entity.Solution
import io.say.better.storage.mysql.domains.solution.entity.SolutionSymbol
import io.say.better.storage.mysql.domains.symbol.entity.Symbol

class SolutionSymbolConverter private constructor() {
    init {
        throw IllegalStateException("Utility class")
    }

    companion object {
        fun toSolutionSymbols(
            savedSolution: Solution,
            symbols: List<Symbol?>,
        ): List<SolutionSymbol> {
            return symbols.stream()
                .map { symbol: Symbol? -> toSolutionSymbol(savedSolution, symbol) }
                .toList()
        }

        private fun toSolutionSymbol(
            solution: Solution,
            symbol: Symbol?,
        ): SolutionSymbol {
            return SolutionSymbol(
                solution = solution,
                symbol = symbol!!,
            )
        }
    }
}
